hayeon/programmers-1/콜라츠-추측.js

```diff
@@ -0,0 +1,32 @@
+function solution(num) {
+  let result = 0;
+
+  for (let i = 0; i <= 500; i++) {
+    if (num !== 1) {
+      num = num % 2 === 0 ? num / 2 : num * 3 + 1;
+    } else {
+      return (result = i);
+    }
+  }
+  return (result = -1);
+}
+
+/*
+  처음 푼 풀이
+  function solution(num) {
+    let result = 0;
+    for (let i = 1; i <= 500; i++) {
+      if (num % 2 === 0) {
+        num % 2;
+      } else {
+        num * 3 + 1;
+      }
+      result += i;
```

Comment:
```suggestion
      result += i;
```
- 이 코드 때문에 체점했을때 단순히 for문 500번 순회하면서 계속 result에 값을 누적해버리니까 아까 모든 테스트 케이스의 결과값이 125250으로 나왔네요 😂

Reply:
저도 풀다보니 왜 누적했는지,,ㅋㅋㅋㅋㅋㅋㅋㅋㅋ 단순히 횟수만 세면 되었던걸 바보같이 누적....🙂